src/v/bytes/details/io_byte_iterator.h

```diff
@@ -43,8 +44,8 @@ class io_byte_iterator {
         }
     }
     io_byte_iterator(
-      io_const_iterator begin,
-      io_const_iterator end,
+      const io_const_iterator& begin,
+      const io_const_iterator& end,
```

Comment:
I'm OK with this, but don't we usually use value and move for sink parameters?

Reply:
yes, but i think for these iterators there isn't anything that is movable, so this avoids one extra copy